### PR TITLE
Location

### PR DESCRIPTION
**Pull request by Flooooooooooorian**
Init Backend for Locations,
Init Frontend for Location,

Not ready

### CODE REVIEW

File: backend/src/main/java/de/neuefische/flooooooooooorian/backend/controller/LocationController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class LocationController {
+
+    private LocationService locationService;
+
+    @Autowired
+    public LocationController(LocationService locationService) {
+        this.locationService = locationService;
+    }
+
+    @GetMapping()
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/flooooooooooorian/backend/controller/LocationController.java:23]:
```suggestion
    @GetMapping
```

File: backend/src/main/java/de/neuefische/flooooooooooorian/backend/model/Location.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Document(collection = "locations")
+@Builder
+public class Location {
+    @Id
+    private String id;
+    private Date creationDate;
+
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/flooooooooooorian/backend/model/Location.java:22]: are those empty lines on purpose?

File: backend/src/main/java/de/neuefische/flooooooooooorian/backend/model/Comment.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+@AllArgsConstructor
+@Document(collection = "comments")
+@Builder
+public class Comment {
+    @Id
+    private String id;
+    private String location_id;
+
+    private User owner;
+    private Date posted_on;
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/flooooooooooorian/backend/model/Comment.java:23]: better use Instant or LocalDateTime

https://i.stack.imgur.com/KVk7O.png

File: backend/src/main/java/de/neuefische/flooooooooooorian/backend/model/Location.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Document(collection = "locations")
+@Builder
+public class Location {
+    @Id
+    private String id;
+    private Date creationDate;
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/flooooooooooorian/backend/model/Location.java:21]: see comment above

File: frontend/src/pages/LocationsPage.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import MapIcon from '@material-ui/icons/Map';
+import ListIcon from '@material-ui/icons/List';
+import {useState} from "react";
+import LocationMap from "../components/LocationMap";
+import styled from "styled-components/macro";
+
+export default function LocationsPage() {
+
+    const locations = useLocations()
+    const [mapIsEnabled, enableMap] = useState(false)
```
Comment (christophersiem) [inline frontend/src/pages/LocationsPage.js:13]:
```suggestion
    const [mapIsEnabled, setMapIsEnabled] = useState(false)
```

File: frontend/src/hooks/useLocations.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+
+export default function useLocations() {
+
+    const [locations, setLocations] = useState([])
+
+    useEffect(() => {
+        axios.get('/api/location')
+            .then((response) => response.data)
+            .then((data) => setLocations(data))
```
Comment (christophersiem) [inline frontend/src/hooks/useLocations.js:12]:
```suggestion
            .then(setLocations)
```

File: backend/src/main/java/de/neuefische/flooooooooooorian/backend/service/LocationService.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import de.neuefische.flooooooooooorian.backend.model.Location;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class LocationService {
+
+    private LocationRepository locationRepository;
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/flooooooooooorian/backend/service/LocationService.java:13]:
```suggestion
    private final LocationRepository locationRepository;
```

File: backend/src/main/java/de/neuefische/flooooooooooorian/backend/controller/LocationController.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/location")
+public class LocationController {
+
+    private LocationService locationService;
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/flooooooooooorian/backend/controller/LocationController.java:16]:
```suggestion
    private final LocationService locationService;
```

**Review by christophersiem — APPROVED** [review]
lgtm, added some suggestions :)

File: backend/src/main/java/de/neuefische/flooooooooooorian/backend/model/User.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+@Builder
+public class User {
+
+    @Id
+    private String id;
+    private String username;
+
+    private List<Location> locations;
+    private List<Picture> pictures;
+    private List<Comment> comments;
```
Comment (fabianschmauder) [inline backend/src/main/java/de/neuefische/flooooooooooorian/backend/model/User.java:25]: do you need this here? then you have duplicate data in your data. Maybe it makes sense to add user id to locations / comments/ pictures and than filter by username when you need all comments from a specific user

File: frontend/src/pages/LocationsPage.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    const locations = useLocations()
+    const [mapIsEnabled, setMapIsEnabled] = useState(false)
+
+    const switchView = (event, value) => {
+        event.preventDefault()
+        setMapIsEnabled(value)
+    }
+
+    return (
+        <div>
```
Comment (fabianschmauder) [inline frontend/src/pages/LocationsPage.js:21]: maybe move both components to separate components? Makes it easier to understand what is happening here

**Review by fabianschmauder — COMMENTED** [review]
LGTM, maybe think about how to structure your db model ;)

**Pull request merged**